### ISSUE TITLE
Make flying-squid server for Minecraft Pocket Edition.

### ISSUE THREAD

**Issue by zxl777** [PrismarineJS/flying-squid#191]
Minecraft Pocket Edition is now popular.
PE has server named Pocketmine, now Pocketmine founders have stopped updating.
I believe node.js speed better than php, please continue to work hard.
thank you.

**Comment by rom1504** [comment]
Yeah this would be interesting. That requires protocol support https://github.com/PrismarineJS/node-minecraft-protocol/issues/235

We're currently working on that with @mhsjlw at https://github.com/mhsjlw/pocket-minecraft-protocol

**Comment by mhsjlw** [comment]
The master plan is to finish [node-raknet](https://github.com/mhsjlw/node-raknet) (a library for helping with Minecraft PE) and finish [pocket-minecraft-protocol](https://github.com/mhsjlw/pocket-minecraft-protocol) and put them into [node-minecraft-protocol](https://github.com/PrismarineJS/node-minecraft-protocol) along with [minecraft-classic-protocol](https://github.com/mhsjlw/minecraft-classic-protocol) to make one library that supports all of those versions, then (instead of turning flying-squid in to a Minecraft PE server or adding Minecraft PE server functionality) we will create a second server software that is specifically for Minecraft PE with a similar JavaScript API to flying-squid. We will still, however, continue updating flying-squid. You can track our progress from those repositories or check the node-minecraft-protocol gitter chat.

**Comment by rom1504** [comment]
I think we could actually consider adding mcpe functionnality into flying-squid, but maybe as a second step indeed since it requires some more stuff (namely https://github.com/PrismarineJS/node-minecraft-protocol/issues/231)

**Comment by mhsjlw** [comment]
My original idea was to keep them separate. Though, that doesn't mean someone couldn't make a plugin or we could add an option to do it.

**Comment by demipixel** [comment]
I would love the option to do one, the other, or both, however how certain issues are handled (i.e. missing block in one, how do you represent that in the other, etc) would need to be up to a plugin. Perhaps allow an option marked either "pc" or "pe". Then allow a plugin that wants to to mark both and handle the issues?

**Comment by rom1504** [comment]
I can't see why you are talking about plugins. I agree building a protocol independant layer would require some work, that's why creating an independant server to start with might be easier, but that protocol independant layer would not be at the plugins layer, it would more likely be just after nmp.
Anyway, we don't even have a basic mcpe server running yet. So let's wait until we have that before actually talking about a mcpc/mcpe server.

**Comment by mhsjlw** [comment]
Yeah, I think we'll close it until we have a working Minecraft PE server model.

**Comment by rom1504** [comment]
hmm why close it ? I think issues should be closed when they are solved. I was just saying there's no need to speculate how we would do a cross-server mcpc/mcpe server right now

**Comment by rom1504** [comment]
pocket-minecraft-protocol basic server is now almost working (ping and packet parsing in general works, currently completing the login sequence) and then we're going at https://github.com/numerous-alpaca/numerous-alpaca

**Comment by mhsjlw** [comment]
any questions about the server software and/or the libraries can be directed to the pocket-minecraft-protocol gitter chat which is here: https://gitter.im/mhsjlw/pocket-minecraft-protocol :wink:
